File: Events.php
<?php

namespace App\Entity;
use App\Repository\EventsRespository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Events
 *
 * @ORM\Table(name="events")
 * * @ORM\Entity(repositoryClass=EventsRespository::class)
 */
class Events
{
    /**
     * @var int
     *
     *
     * @ORM\Column(name="id_event", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idEvent;

    /**
     * @var string

     * @ORM\Column(name="nom_event", type="string", length=99, nullable=false)
    *  @Assert\NotBlank(message="Remplir le nom")
     */
    private $nomEvent;

    /**
     * @var \DateTime

     * @ORM\Column(name="date_deb", type="date", nullable=false)
     */
    private $dateDeb;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_fin", type="date", nullable=false)
     */
    private $dateFin;

    /**
     * @var int

     * @ORM\Column(name="nbr_place", type="integer", nullable=false)
     *  @Assert\NotBlank(message="Remplir le nbr de place")

     */
    private $nbrPlace;

    /**
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="events")
     */
    private $categorie;

    public function getIdEvent(): ?int
    {
        return $this->idEvent;
    }

    public function getNomEvent(): ?string
    {
        return $this->nomEvent;
    }

    public function setNomEvent(string $nomEvent): self
    {
        $this->nomEvent = $nomEvent;

        return $this;
    }

    public function getDateDeb(): ?\DateTimeInterface
    {
        return $this->dateDeb;
    }

    public function setDateDeb(\DateTimeInterface $dateDeb): self
    {
        $this->dateDeb = $dateDeb;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(\DateTimeInterface $dateFin): self
    {
        $this->dateFin = $dateFin;

        return $this;
    }

    public function getNbrPlace(): ?int
    {
        return $this->nbrPlace;
    }

    public function setNbrPlace(int $nbrPlace): self
    {
        $this->nbrPlace = $nbrPlace;

        return $this;
    }

    public function getCategorie(): ?Categorie
    {
        return $this->categorie;
    }

    public function setCategorie(?Categorie $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }


}

File: src/Entity/Events.java
package Entity;


import java.time.LocalDate;

public class Events {

    private int id_event;

    private String nom_event;
    private LocalDate date_deb;
    private LocalDate date_fin;

    public int getId_event() {
        return id_event;
    }

    public void setId_event(int id_event) {
        this.id_event = id_event;
    }

    public String getNom_event() {
        return nom_event;
    }

    public void setNom_event(String nom_event) {
        this.nom_event = nom_event;
    }

    public LocalDate getDate_deb() {
        return date_deb;
    }

    public void setDate_deb(LocalDate date_deb) {
        this.date_deb = date_deb;
    }

    public LocalDate getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(LocalDate date_fin) {
        this.date_fin = date_fin;
    }

    public Events() {
    }

    public Events(int id_event, String nom_event, LocalDate date_deb, LocalDate date_fin) {
        this.id_event = id_event;
        this.nom_event = nom_event;
        this.date_deb = date_deb;
        this.date_fin = date_fin;
    }

    @Override
    public String toString() {
        return "Events{" + "id_event=" + id_event + ", nom_event=" + nom_event + ", date_deb=" + date_deb + ", date_fin=" + date_fin + '}';
    }





}














